flocker/control/test/test_protocol.py

```diff
@@ -451,3 +518,295 @@ class FakeAgentInterfaceTests(iconvergence_agent_tests_factory(
+            arg1=object(),
+            arg2=object()
+        )
+        actual_result = client.send_request(server, **expected_results)
+
+        self.assertEqual(expected_results, actual_result)
+
+    def test_decorated_name(self):
+        """
+        ``with_eliot_context`` returns a decorator function with the same name
+        as the decorated function.
+        """
+        self.assertEqual(
+            'handle_request',
+            ServerProcess().handle_request.__name__
+        )
+
+    def test_decorated_docstring(self):
+        """
+        ``with_eliot_context`` returns a decorator function with the same
+        docstring as the decorated function.
+        """
+        self.assertEqual(
+            'Handle request.',
+            ServerProcess().handle_request.__doc__.strip()
+        )
+
+    def test_positional_arguments_error(self):
+        """
+        The decorator returned by ``with_eliot_context`` does not accept
+        positional arguments, regardless of whether the decorated function
+        accepts them.
+        """
+        server = ServerProcess()
+        dummy_eliot_context = object()
+        error = self.assertRaises(
+            TypeError,
+            server.handle_request, dummy_eliot_context, 'arg1', 'arg2'
+        )
+        self.assertEqual(
+            'responder() takes exactly 2 arguments (4 given)',
+            str(error)
+        )
+
+
+class ClusterStatusCommandTests(SynchronousTestCase):
+    """
+    Tests for ``ClusterStatusCommand``.
+    """
+    def test_command_arguments(self):
+        """
+        ClusterStatusCommand requires the following arguments.
+        """
+        self.assertEqual(
+            sorted(['configuration', 'state', 'eliot_context']),
+            sorted(v[0] for v in ClusterStatusCommand.arguments))
+
+
+class AgentLocatorTests(SynchronousTestCase):
+    """
+    Tests for ``_AgentLocator``.
+    """
+    @validate_logging(None)
+    def test_logger(self, logger):
+        """
+        ``_AgentLocator.logger`` is a property that returns the ``logger``
+        attribute of the ``Agent`` supplied to its initialiser.
+        """
+        fake_agent = FakeAgent()
+        self.patch(fake_agent, 'logger', logger)
+        locator = _AgentLocator(agent=fake_agent)
+        self.assertIs(logger, locator.logger)
+
+
+class ClusterUpdatedTests(SynchronousTestCase):
+    """
+    Tests for the responder for ``ClusterStatusCommand``.
+    """
+    @validate_logging(None)
+    def test_responder_logging(self, logger):
+        """
+        ``cluster_updated`` is decorated using ``with_eliot_context`` and
+        therefore requires an eliot_context argument. The supplied
+        eliot_context is used as the context for messages logged in that
+        method.
+        """
+        fake_agent = FakeAgent()
+        self.patch(fake_agent, 'logger', logger)
+        locator = _AgentLocator(agent=fake_agent)
+        with SEND_REQUEST(logger) as action:
```

Comment:
Same issue as above: you need to move `cluster_updated` call out of the action context.